queries/elixir/highlights.scm

```diff
@@ -11,144 +16,184 @@
   "}"
 ] @punctuation.bracket
 
+; Operators
 [
-  ","
-  "->"
+  "!"
+  "!="
+  "!=="
+  "&"
+  "&&"
+  "&&&"
+  "*"
+  "**"
+  "+"
+  "++"
+  "+++"
+  "-"
+  "--"
+  "---"
   "."
-] @punctuation.delimiter
-
-[
-  (sigil_start)
-  (sigil_end)
-  (heredoc_start)
-  (heredoc_end)
-] @punctuation.special
-
-(interpolation ["#{" "}"] @punctuation.special)
-
+  ".."
+  "/"
+  "//"
+  "::"
+  "<"
+  "<-"
+  "<<<"
+  "<<~"
+  "<="
+  "<>"
+  "<|>"
+  "<~"
+  "<~>"
+  "="
+  "=="
+  "==="
+  "=>"
+  "=~"
+  ">"
+  ">="
+  ">>>"
+  "@"
+  "^"
+  "and"
+  "in"
+  "not"
+  "or"
+  "when"
+  "|"
+  "|>"
+  "||"
+  "|||"
+  "~>"
+  "~>>"
+  "~~~"
+  "~"
+] @operator
+
+; Do/Fn blocks
 [
-  "after"
-  "and" 
-  "catch"
   "do"
-  "else"
   "end"
   "fn"
-  "in"
-  "not in"
-  "not"
-  "or"
-  "rescue"
-  "when" 
 ] @keyword
 
+; Exception blocks
 [
-  (comment) 
-  (unused_identifier)
-] @comment
-
-[
-  (heredoc_content)
-  (sigil_content)
-  (string)
-] @string
-
-; __MODULE__ and friends
-(special_identifier) @constant.builtin
-
-(map ["%{" "}"] @constructor)
+  "catch"
+  "rescue" 
+] @exception
 
-(struct ["%" "{" "}"] @constructor)
+; Conditional blocks
+[ 
+  "else" 
+] @conditional
 
-(binary_op operator: _ @operator)
+; Comments
+(comment) @comment
 
-(unary_op operator: _ @operator)
+; Parser errors
+(ERROR) @error
 
-(atom) @symbol
+; Sigils
+(sigil (sigil_name) @operator quoted_start: _ @operator quoted_end: _ @operator)
 
-(keyword) @parameter
+; Modules
+(alias) @type
 
+; Atoms
 [
-  (true) 
-  (false)
-] @boolean
-
-(nil) @constant.builtin
+  (atom)
+  (quoted_atom) 
+] @symbol
 
-(sigil) @string.special
+; Keywords
+(keyword) @symbol
 
-(identifier) @variable
+; Strings
+(string) @string
 
-(module) @type
+; Interpolation
+(interpolation "#{" @string.escape (_)? "}" @string.escape)
 
-(function_identifier) @function
+; Escape sequences
+(escape_sequence) @string.escape
 
+; Integers
 (integer) @number
 
+; Floats
 (float) @float
 
+; Characters
 [
-  (sigil_start) 
-  (sigil_end)
-] @string.special
-
-; Module attributes as "attributes"
-(unary_op operator: "@" @attribute [
-  (call function: (function_identifier) @attribute) 
-  (identifier) @attribute
+  (char)
+  (charlist)
+] @character
+
+; Booleans
+(boolean) @boolean
+
+; Nil
+(nil) @constant.builtin
+
+; Calling a local function
+(call target: (identifier) @function) 
```

Comment:
There's an additional edge with the pipe operator, if we have `a |> b`, then `b` is a function call. However in `def a |> b`, `b` is an identifier. You may want to special case it [like this](https://github.com/elixir-lang/tree-sitter-elixir/blob/7ae20df181b86c79d826abd5aec7a3e32e3d8438/queries/highlights.scm#L140-L152).